File: apps/api/src/routes/security.ts
import { Router, type Request, type Response } from "express";
import { createCsrfToken, setCsrfCookie } from "../middleware/security.js";

const router: Router = Router();

router.get("/csrf-token", (_req: Request, res: Response) => {
  const csrfToken = createCsrfToken();
  setCsrfCookie(res, csrfToken);
  res.setHeader("cache-control", "no-store");
  res.json({ csrfToken });
});

export { router as securityRouter };

File: services/api/routes/security.py
from fastapi import APIRouter
from starlette.responses import JSONResponse

from middleware.csrf import create_csrf_token, set_csrf_cookie

router = APIRouter(prefix="/api/security")


@router.get("/csrf-token")
async def csrf_token():
    token = create_csrf_token()
    response = JSONResponse(content={"csrfToken": token})
    set_csrf_cookie(response, token)
    response.headers["cache-control"] = "no-store"
    return response
